### PR TITLE
Elaborate on tx handling in "Understanding mining" guide

### PR DESCRIPTION
**Pull request by cburgdorf**
### What was wrong?

Mining guide in docs didn't demonstrate how to include transactions.

### How was it fixed?

Added another paragraph that demonstrates how to include a simple transaction.

This PR also includes two another change that is *driven* by the docs and hence included in this PR.

I added a `get_transaction_nonce` API to the `VM` class. Reasoning is that it feels not very ergonomic to drill into `vm.state.account_db` in order to get a nonce. I also considered setting the `nonce` *implicitly* as part of the `create_unsigned_transaction` API if no `nonce` is passed. However, I believe there are still scenarios where one needs to retrieve the `nonce` directly which brings me back to the point that drilling into `vm.state.account_db` feels not very ergonomic (and may even straight violate the general principle of encapsulation).

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

File: docs/guides/evm/understanding_the_mining_process.rst [hunk 1]
```diff
@@ -210,4 +210,169 @@ information that we need to calculate the ``nonce`` and the ``mix_hash``.
...
+In practice, we have to pay the miners to put our transaction in a block. However, there is no
+*technical* reason why we have to pay for the computing power, but only an economical, i.e. in reality
+we'll usually have trouble finding a miner who's willing to include a transaction that doesn't pay
+for its computational costs.
+
+In this example however, **we are the miner** which means we are free to include any transactions
+we like. In the spirit of this guide, let's start simple and create a transactions that sends zero
+ether from one address to another address. Keep in mind that even if the value being transferred
+is zero, there's still a computational cost for the processing but since we are the miner, we'll
+mine it anyway even if noone is willing to pay for it!
```
Comment (dylanjw) [inline docs/guides/evm/understanding_the_mining_process.rst]: *no one

File: docs/guides/evm/understanding_the_mining_process.rst [hunk 2]
```diff
@@ -210,4 +210,169 @@ information that we need to calculate the ``nonce`` and the ``mix_hash``.
...
+    SENDER_PRIVATE_KEY = keys.PrivateKey(
+      decode_hex('0x45a915e4d060149eb4365960e6a7a45f334393093061116b197e3240065ff2d8')
+    )
+
+    SENDER = Address(SENDER_PRIVATE_KEY.public_key.to_canonical_address())
+
+    RECEIVER = Address(b'\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\x02')
+
+One thing that strikes out here is that we only need the plain address for the receiver whereas for
+the sender we are obtaining an adressed derived from the ``SENDER_PRIVATE_KEY``. That's because we
```
Comment (dylanjw) [inline docs/guides/evm/understanding_the_mining_process.rst]: *address

**Comment by cburgdorf** [comment]
@dylanjw @hwwhww Thanks for the corrections! Fixed!

@pipermerriam are you ok with the added API (well, and the doc change)

File: evm/vm/base.py [hunk 3]
```diff
@@ -588,6 +596,12 @@ def previous_hashes(self):
     #
     # Transactions
     #
+    def get_transaction_nonce(self, sender: Address) -> int:
```
Comment (pipermerriam) [inline evm/vm/base.py]: As we talked about, lets drop this API.

**Pull request merged**